server/src/services/websocket-server.ts

```
@@ -0,0 +1,461 @@
+            }
+        });
+
+        // Handle top-level WebSocket close/error (covers cases before/during auth)
+        // For connections that have completed authentication we do **not** destroy
+        // the session immediately – instead we schedule orphan cleanup so the
+        // container can be resumed within the RESUME_GRACE_MS window.
+        const topLevelCloseHandler = (code: number, reason: Buffer) => {
+            log(`[Server] WebSocket closed. Code: ${code}, Reason: ${reason.toString()}`);
+
+            const existing = getSessions().get(sessionId);
+            if (existing && existing.authenticated) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)